pywatts/summaries/tsne_visualisation.py

```diff
@@ -0,0 +1,134 @@
+from typing import Dict
+
+import matplotlib.pyplot as plt
+import tikzplotlib
+import xarray as xr
+from pywatts.core.base_summary import BaseSummary
+from sklearn.manifold import TSNE
+import numpy as np
+
+from pywatts.core.summary_object import SummaryObjectList
+
+
+class TSNESummary(BaseSummary):
+    """
+    Summary that performs a TSNE to visualise the data. It is possible to specify mask by providing kwargs
+    to the transform method that end with _masked.
+    :param max_point: The maximum number of points per data that should be plotted
+    :type max_point: int
+    :param all_in_one_plot: Flag indicating if all input data should be visualised in the same plot. If not the column
+                            GT is visualised with all other in separate plots.
+    :type all_in_one_plot: Boolean
+    :param tsne_params: Params for the TSNE Visualisation. See sklearn.
+    :type tsne_params: Dict
+    """
+
+    def __init__(self, name="TSNE",
+                 max_points=10000, all_in_one_plot=False, tsne_params=None):
+        super().__init__(name)
+        self.tsne_params = {} if tsne_params is None else tsne_params
+        self.max_points = max_points
+        self.all_in_one_plot = all_in_one_plot
+
+    def get_params(self) -> Dict[str, object]:
+        """"
+        Get the params of the TSNE Summary
+        :return: A dict containing all parameter of the tsne_summary
+        :rtype: Dict
+        """
+        return {
+            "tsne_params": self.tsne_params,
+            "max_points": self.max_points,
+            "all_in_one_plot": self.all_in_one_plot
+        }
+
+    def set_params(self, max_points=None, all_in_one_plot=None, tsne_params=None):
+        """
+        Set the params of the tsne summary.
+        :param max_point: The maximum number of points per data that should be plotted
+        :type max_point: int
+        :param all_in_one_plot: Flag indicating if all input data should be visualised in the same plot. If not the column
+                                GT is visualised with all other in separate plots.
+        :type all_in_one_plot: Boolean
+        :param tsne_params: Params for the TSNE Visualisation. See sklearn.
+        :type tsne_params: Dict
+        """
+        if max_points is not None:
+            self.max_points = max_points
+        if all_in_one_plot is not None:
+            self.all_in_one_plot = all_in_one_plot
+        if tsne_params is not None:
+            self.tsne_params = tsne_params
+
+    def transform(self, file_manager, gt: xr.DataArray, **kwargs: xr.DataArray):
+        """
+        Calculates the TSNE and visualise it. kwargs that end with _masked are masks for the other input. E.g. gt_masked
+        is a mask for gt.
```

Comment:
Calculates the TSNE and visualises it. kwargs that end with _masked are masks for the other input. For example, gt_masked is a mask for gt.